internal/webhook/template_webhook.go

```diff
@@ -55,7 +64,31 @@ func (*ClusterTemplateValidator) ValidateUpdate(_ context.Context, _ runtime.Obj
 }
 
 // ValidateDelete implements webhook.Validator so a webhook will be registered for the type.
-func (*ClusterTemplateValidator) ValidateDelete(_ context.Context, _ runtime.Object) (admission.Warnings, error) {
+func (v *ClusterTemplateValidator) ValidateDelete(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	template, ok := obj.(*v1alpha1.ClusterTemplate)
+	if !ok {
+		return admission.Warnings{"Wrong object"}, apierrors.NewBadRequest(fmt.Sprintf("expected ClusterTemplate but got a %T", obj))
+	}
+
+	managedClusters := &v1alpha1.ManagedClusterList{}
+	listOptions := client.ListOptions{
+		FieldSelector: fields.SelectorFromSet(fields.Set{v1alpha1.TemplateKey: template.Name}),
+		Limit:         1,
+	}
+	err := v.Client.List(ctx, managedClusters, &listOptions)
+	if err != nil {
+		if err.Error() == "Index with name field:.spec.template does not exist" {
+			// ManagedCluster may not be indexed because of the incomplete reconciliation
+			log.FromContext(ctx).Error(err, "ClusterTemplateValidator")
+			return nil, nil
+		}
```

Comment:
Is it correct that we allow the `ClusterTemplate` deletion in this case? What if this error occurred but there are still `ManagedClusters` that should prevent the template from being removed? I think it should be handled in a different way.

Reply:
We may have a workaround to remove the templates iterating over all ManagedClusters.
In fact, I'l not sure what we can do if we have a half-initialized system (where the problem occurred somewhere between template creation and proper initialization of a Manager).